src/irmin/proof.ml

```diff
@@ -292,9 +292,21 @@ struct
 
   let dehydrate_stream_node h v =
     let v : P.pre_elt =
-      match N.to_pre_elt v with
+      match N.to_elt v with
       | `Node l -> P.Node l
-      | `Inode (length, proofs) -> P.Inode { length; proofs }
+      | `Inode (length, proofs) ->
+          let proofs =
```

Comment:
could we avoid this? e.g. assume that the backend returns a proper elt (with extenders) but still try to compress the stream if the backend hasn't done it correctly.

Alternativetly just replace `int list` by `int` in `Node.elt`...